starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -82,7 +83,11 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    Inventory deleted = this.mongoTemplate.findById(id, Inventory.class);
+    if(deleted == null){return null;}
```

Comment:
For the return value, it won't let me return a non-optional, but if the value is null, I cannot using Optional.of(), as that throws an error. This being said, because findandremove returns an Inventory item (non-optional) or null, I also can't set Option.of() to the result of findandremove, and I cannot set an optional variable to be equal to the result of findandremove, so I still have a quick check if the result of findandremove is null. I will keep looking however. Also, I know that Optional<Inventory> can be null or an Inventory object, but I get an error saying that I need to return an optional type, not inventory, which is why I am doing the check still.